pkg/ring/ring_test.go

```diff
@@ -447,26 +454,29 @@ func TestRing_Get_ZoneAwareness(t *testing.T) {
 				set, err = ring.Get(testValues[i], Write, instances, bufHosts, bufZones)
 				if testData.expectedErr != "" {
 					require.EqualError(t, err, testData.expectedErr)
+					continue // Skip the rest of the assertions if we were expecting an error.
 				} else {
 					require.NoError(t, err)
 				}
 
-				// Skip the rest of the assertions if we were expecting an error.
-				if testData.expectedErr != "" {
-					continue
-				}
-
 				// Check that we have the expected number of instances for replication.
 				assert.Equal(t, testData.expectedInstances, len(set.Instances))
 
 				// Ensure all instances are in a different zone (only if zone-awareness is enabled).
 				if testData.zoneAwarenessEnabled {
-					zones := make(map[string]struct{})
+					zones := make(map[string]int)
+					maxNumOfHostsPerZone := math.Ceil(float64(testData.replicationFactor) / float64(testData.numZones))
+
 					for i := 0; i < len(set.Instances); i++ {
-						if _, ok := zones[set.Instances[i].Zone]; ok {
-							t.Fatal("found multiple instances in the same zone")
+						if _, ok := zones[set.Instances[i].Zone]; !ok {
+							zones[set.Instances[i].Zone] = 1
+						} else {
+							zones[set.Instances[i].Zone]++
+
+							if zones[set.Instances[i].Zone] > int(maxNumOfHostsPerZone) {
+								t.Fatal("instances not spread across zones evenly")
```

Comment:
Can we add assertions to also make sure when RF > zones, each zone can get at least 1 instance?

Reply:
Sure!